Extractor/History/HistoryScheduler.cs

```diff
@@ -505,14 +520,27 @@ private void HistoryDataHandler(HistoryReadNode node)
             var last = DateTime.MinValue;
             var first = DateTime.MaxValue;
 
-            if (dataPoints.Count != 0)
+            var goodDataPoints = dataPoints.Where(dp => StatusCode.IsGood(dp.StatusCode));
+
+            int goodDpCount = 0;
+
+            foreach (var dp in goodDataPoints)
             {
-                (first, last) = dataPoints.MinMax(dp => dp.SourceTimestamp);
+                if (first > dp.SourceTimestamp)
+                {
+                    first = dp.SourceTimestamp;
+                }
+
+                if (last < dp.SourceTimestamp)
+                {
+                    last = dp.SourceTimestamp;
+                }
+                goodDpCount++;
```

Comment:
Is first-last point determination based on only good points? The change block above does filter out unwanted points depending on the configuration, do we want to re-scope the points here?

Reply:
I think it has to be. Some servers will return a bunch of bad datapoints if they are in a bad state somehow. We can't really trust those when it comes to state.

The exception is if we rely on timestamps for pagination, in which case we have to trust it, but since that's a workaround anyway it's fine.